kci_build

```diff
@@ -401,6 +403,23 @@ class cmd_push_kernel(Command):
         install = kernelci.build.Step.get_install_path(args.kdir, args.output)
         meta = kernelci.build.Metadata(install)
         publish_path = meta.get('bmeta', 'kernel', 'publish_path')
+        defconfig = meta.get('bmeta', 'kernel', 'defconfig')
+        # gki_defconfig is exceptional and generate hundreds of megabytes
+        # several files, which require compression
+        # to workaround current upload limits
+        if defconfig == "gki_defconfig":
+            kernel_image_name = meta.get('bmeta', 'kernel', 'image')
+            compressed_artifacts = ["logs/kernel.log"]
+            compressed_artifacts.append('/'.join(["kernel",
+                                        kernel_image_name]))
+            for e in compressed_artifacts:
+                source_filename = os.path.join(install, e)
+                data = open(source_filename, 'rb').read()
+                zf = xz.open(f'{source_filename}.xz', mode='wb',
+                             format=xz.FORMAT_XZ, preset=9)
+                zf.write(data)
+                zf.close()
+                os.unlink(source_filename)
```

Comment:
This may be a follow up, but it appears to me as this could be extracted to a separate function